python/ray/util/spark/cluster_init.py

```diff
@@ -522,7 +481,11 @@ def _init_ray_cluster(
 
     _logger.info("Ray head node started.")
 
-    start_hook.on_ray_dashboard_created(ray_dashboard_port)
+    if include_dashboard is None or include_dashboard is True:
+        # If include_dashboard is None, we don't set `--include-dashboard` option,
+        # in this case Ray will decide whether dashboard can be started
+        # (e.g. checking any missing dependencies).
+        start_hook.on_ray_dashboard_created(ray_dashboard_port)
```

Comment:
if `include_dashboard` is None, it's not guaranteed that the dashboard is created.

If you want to know whether dashboard is started, you can check the return value of `ray.init().address_info["webui_url"]`